django-stubs/utils/functional.pyi

```diff
@@ -43,6 +43,7 @@ class _StrPromise(Promise, Sequence[str]):
     # Mypy requires this for the attribute hook to take effect
     def __getattribute__(self, __name: str) -> Any: ...
 
+_StrOrPromise = Union[str, _StrPromise]
```

Comment:
Maybe `_StrPromise` should be public? Since it'll be exposed to users, right? 

So that one can do `str | StrPromise` if passing around/accepting `gettext_lazy` values

```suggestion
_StrOrPromise = Union[str, StrPromise]
```